services/director-v2/tests/integration/02/utils.py

```diff
@@ -198,45 +150,46 @@ async def assert_start_service(
     }
 
     result = await director_v2_client.post(
-        "/dynamic_services", json=data, headers=headers, allow_redirects=False
+        "/dynamic_services", json=data, headers=headers, allow_redirects=True
     )
-    result = await handle_307_if_required(director_v2_client, director_v0_url, result)
-    assert result.status_code == 201, result.text
+    assert result.status_code == httpx.codes.CREATED, result.text
 
 
 async def get_service_data(
     director_v2_client: httpx.AsyncClient,
-    director_v0_url: URL,
     service_uuid: str,
     node_data: Node,
 ) -> Dict[str, Any]:
-    result = await director_v2_client.get(
+
+    # result =
+    response = await director_v2_client.get(
         f"/dynamic_services/{service_uuid}", allow_redirects=False
     )
-    result = await handle_307_if_required(director_v2_client, director_v0_url, result)
-    assert result.status_code == 200, result.text
-
-    payload = result.json()
+    if response.status_code == httpx.codes.TEMPORARY_REDIRECT:
+        # NOTE: so we have a redirect, and it seems the director_v2_client does not like it at all
+        #  moving from the testserver to the director in this GET call
+        # which is why we use a DIFFERENT httpx client for this... (sic).
+        # This actually works well when running inside the swarm... WTF???
```

Comment:
Not sure about what you mean on this last line.

Reply:
yes this comes from your comments.